src/App.tsx

```diff
@@ -1,8 +1,72 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { Good, GoodsData } from './types';
+import { getGoods } from './API/API';
+import { GoodsList } from './Components/GoodsList/GoodsList'
+import { Button } from './Components/Button/Button';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+};
 
-export default App;
+export class App extends Component<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  getAllGoods = () => {
+    getGoods()
+      .then(({ data }: GoodsData) => this.setState({
+        goods: data,
+      }))
+      .catch(error => {
+        throw new Error(error.message);
+      });
```

Comment:
```suggestion
      .catch(error => {
        console.worn(error.message)
      });
```

поэкспериментируй с этим. сейчас ты отлавливаешь ошибку и создаешь новую, а новую не отлавливаешь